File: VTI_Galaxy_BE/src/main/java/demo/services/interfaceClass/AccountService.java
package demo.services.interfaceClass;

import demo.modal.dto.AccountDto;
import demo.modal.request.AccountRequest;

import java.util.List;

public interface AccountService {
    // get
    List<AccountDto> getAccounts();
    AccountDto getAccountById(int id);

    // post
    AccountDto registerAdmin(AccountRequest request);
    AccountDto registerUser(AccountRequest request);

    //put
    AccountDto updateAccount(AccountRequest request, int id);
    AccountDto updateEmail(int id, String email);
    AccountDto updatePassword(int id, String password);

    // delete
    AccountDto deleteAccount(int id);
}

File: VTI_Glaxy_FE/src/services/accountService.js
import axiosClient from "./axiosClient"

const accountService = {
   fetchAccounts(){
      return axiosClient.get("/getAccounts")
   },
   fetchAccountById(accountId){
      return axiosClient.get(`/getAccountById?accountId=${accountId}`)
   },
   createAccountAdmin(accountRequest){
      return axiosClient.post("/registerAdmin", accountRequest)
   },
   createAccountUser(accountRequest){
      return axiosClient.post("/registerUser", accountRequest)
   },
   updateAccount(accountId, accountRequest){
      return axiosClient.put(`/putAccount?accountId=${accountId}`, accountRequest)
   },
   updateEmail(accountId, email){
      return axiosClient.put(`/putEmail?accountId=${accountId}&email=${email}`)
   },
   updatePassword(accountId, password){
      return axiosClient.put(`/putPassword?accountId=${accountId}&password=${password}`)
   },
   deleteAccount(accountId){
      return axiosClient.put(`/deleteAccount?accountId=${accountId}`)
   }
}

export default accountService